ProcessLib/ThermoHydroMechanics/CreateThermoHydroMechanicsProcess.cpp

```diff
@@ -0,0 +1,311 @@
+    {
+        OGS_FATAL(
+            "Pressure process variable '%s' is not a scalar variable but has "
+            "%d components.",
+            process_variables[1].get().getName().c_str(),
+            process_variables[1].get().getNumberOfComponents(),
+            DisplacementDim);
+    }
+
+    DBUG("Associate temperature with process variable \'%s\'.",
+         process_variables[0].get().getName().c_str());
+    if (process_variables[0].get().getNumberOfComponents() != 1)
+    {
+        OGS_FATAL(
+            "Pressure process variable '%s' is not a scalar variable but has "
+            "%d components.",
+            process_variables[0].get().getName().c_str(),
+            process_variables[0].get().getNumberOfComponents(),
+            DisplacementDim);
+    }
+
+    // Constitutive relation.
+    // read type;
+    auto const constitutive_relation_config =
+        //! \ogs_file_param{prj__processes__process__THERMO_HYDRO_MECHANICS__constitutive_relation}
+        config.getConfigSubtree("constitutive_relation");
+
+    auto const type =
+        //! \ogs_file_param{prj__processes__process__THERMO_HYDRO_MECHANICS__constitutive_relation__type}
+        constitutive_relation_config.peekConfigParameter<std::string>("type");
+
+    std::unique_ptr<MaterialLib::Solids::MechanicsBase<DisplacementDim>>
+        material = nullptr;
+    if (type == "LinearElasticIsotropic")
+    {
+        material =
+            MaterialLib::Solids::createLinearElasticIsotropic<DisplacementDim>(
+                parameters, constitutive_relation_config);
+    }
+    else if (type == "Ehlers")
+    {
+        material =
+            MaterialLib::Solids::Ehlers::createEhlers<DisplacementDim>(
+                parameters, constitutive_relation_config);
+    }
+    else if (type == "Lubby2")
+    {
+        material =
+            MaterialLib::Solids::Lubby2::createLubby2<DisplacementDim>(
+                parameters, constitutive_relation_config);
+    }
+    else
+    {
+        OGS_FATAL(
+            "Cannot construct constitutive relation of given type \'%s\'.",
+            type.c_str());
+    }
+
+    // Intrinsic permeability
+    auto& intrinsic_permeability = findParameter<double>(
+        config,
+        //! \ogs_file_param_special{prj__processes__process__THERMO_HYDRO_MECHANICS_intrinsic_permeability}
+        "intrinsic_permeability", parameters, 1);
+
+    DBUG("Use \'%s\' as intrinsic permeability parameter.",
+         intrinsic_permeability.name.c_str());
+
+    // Storage coefficient
+    auto& storage_coefficient = findParameter<double>(
+        config,
+        //! \ogs_file_param_special{prj__processes__process__THERMO_HYDRO_MECHANICS_storage_coefficient}
+        "storage_coefficient", parameters, 1);
+
+    DBUG("Use \'%s\' as storage coefficient parameter.",
+         storage_coefficient.name.c_str());
+
+    // Fluid viscosity
+    auto& fluid_viscosity = findParameter<double>(
```

Comment:
suggest to use fluid property class (see PR. #1798: https://github.com/ufz/ogs/pull/1798/files#diff-c6c528a852ac448ad3706c2e71b24b65R104)